eng/Subsets.props

```diff
@@ -66,6 +66,7 @@
 
     <_subsetCategory Condition="'$(SubsetCategory)' != ''">$(SubsetCategory.ToLowerInvariant())</_subsetCategory>
     <_subsetCategory Condition="'$(SubsetCategory)' == ''">$(DefaultSubsetCategories)</_subsetCategory>
+    <_subsetCategory Condition="'$(SubsetCategory)' == '' and '$(TargetOS)' == 'iOS'">libraries-mono</_subsetCategory>
```

Comment:
Should we error if `SubsetCategory` is not empty and contains either installer or coreclr when `TargetOS` is `iOS`?

Reply:
I feel like if you manually override subset category that way you deserve to get the failures :)

Reply:
I get that, but it is good to help developers get readable failures? Sometimes because of the lack of those validations we get a lot of questions, why this doesn't work, why am I getting this error, etc? We already have a target that does validation for the subset categories, if I remember correctly.

Reply:
I think it's ok to set RuntimeFlavor based on the passed in `SubsetCategory` but I don't think that `SubsetCategory` should be inferred from `TargetOS`. The `SubsetCategory` can also just contain `mono`, or `mono-libraries-installer` therefore I think it's necessary to be intentional here and require the category to be passed in.

Reply:
@ViktorHofer why not? `coreclr` or `installer` are not valid subset categories when you have TargetOS=iOS so having a default when nothing is set explicitly makes a lot of sense.

Reply:
@safern I agree with your point, it will be a lot easier to diagnose if you accidentally pass an invalid category so I'll add some validation.

Reply:
Because until now we have never calculated SubsetCategories based on other inputs. Why would you always default to mono-libraries? Why not just mono? Why not mono-libraries-installer? Why are you choosing exactly `mono-libraries`?

Reply:
The whole point of the Subset-Feature is that devs have a hook to only build segments of the repository **if they wish to do so**. The default should always be the full repository, minus the parts that are unsupported in that configuration. If you really want to provide a default (which I don't think is a good idea as said) then you should add `-installer` to the subsetcategory. At the end of the day it's installer who will build the shared framework packages and presumably also the iOS target packages.

Reply:
I thought about this more. I'm fine with providing a default value for the subset categories when a configuration is passed in that isn't supported on one of those categories but I would not want to exclude installer which brings me back to my question: Why did you choose `mono-libraries` and not `mono-libraries-installer`?

Reply:
@ViktorHofer installer doesn't make sense at all, there won't be an installer for the iOS bits. The default of `mono-libraries` is everything that we want to build.

Reply:
I can move the change to set `DefaultSubsetCategories` for `TargetOS==iOS` instead if that helps making it clearer.

Reply:
I discussed this with Viktor offline and there was some misunderstanding. `installer` is not just for building the .deb/.msi/.pkg installers but also does other things like creating runtime packs and ref packs. So we might eventually need it for iOS too.

Reply:
Regarding the subset validation: Viktor and me took a look and it seems a little more complicated to add than expected. We'll do it in a separate PR.